src/portal/src/app/shared/components/navigator/navigator.component.ts

```diff
@@ -199,16 +208,43 @@ export class NavigatorComponent implements OnInit {
 
     // Log out system
     logOut(): void {
-        // Naviagte to the sign in router-guard
-        // Appending 'signout' means destroy session cache
-        let signout = true;
-        let redirect_url = this.location.pathname;
-        let navigatorExtra: NavigationExtras = {
-            queryParams: { signout, redirect_url },
-        };
-        this.router.navigate([CommonRoutes.EMBEDDED_SIGN_IN], navigatorExtra);
-        // Confirm search result panel is close
-        this.searchTrigger.closeSearch(true);
+        // Call the service to send out the http request
+        this.session.signOff().subscribe(
+            () => {
+                // Naviagte to the sign in router-guard
+                // Appending 'signout' means destroy session cache
+                let signout = true;
+                let redirect_url = this.location.pathname;
+                let navigatorExtra: NavigationExtras = {
+                    queryParams: { signout, redirect_url },
+                };
+                this.router.navigate(
+                    [CommonRoutes.EMBEDDED_SIGN_IN],
+                    navigatorExtra
+                );
+                // Confirm search result panel is close
+                this.searchTrigger.closeSearch(true);
+            },
+            error => {
+                // 403 oidc logout no body;
+                if (this.isOidcLoginMode && error && error.status === 403) {
```

Comment:
Does it mean under oidc auth mode, the UI can not handler the 403 error other than redirect?

Reply:
this actually a workaround for oidc logout since we disable the render method in the base handler.